drivers/serial/uart_mcux.c

```diff
@@ -289,6 +294,9 @@ static const struct uart_mcux_config uart_mcux_0_config = {
 	.clock_name = DT_UART_MCUX_0_CLOCK_NAME,
 	.clock_subsys = (clock_control_subsys_t)DT_UART_MCUX_0_CLOCK_SUBSYS,
 	.baud_rate = DT_UART_MCUX_0_BAUD_RATE,
+#if DT_UART_MCUX_0_HW_FLOW_CONTROL == 1
```

Comment:
Actually why do you need the #if at all?

seems like:

`.hw_flow_control = DT_UART_MCUX_0_HW_FLOW_CONTROL,`

is fine since you'll get 0 or 1 depending on the property existing or not.